backend/controllers/attendanceStatistics.js

```diff
@@ -1,126 +1,247 @@
+  const activeCaseLoadId = activeCaseLoad ? activeCaseLoad.caseLoadId : null
+
+  return {
+    activeCaseLoad,
+    inactiveCaseLoads,
+    activeCaseLoadId,
+  }
+}
+
+const absentReasonTableViewModel = offenderData => ({
+  sortOptions: [
+    { value: '0_ascending', text: 'Name (A-Z)' },
+    { value: '0_descending', text: 'Name (Z-A)' },
+    { value: '2_ascending', text: 'Location (A-Z)' },
+    { value: '2_descending', text: 'Location (Z-A)' },
+    { value: '3_ascending', text: 'Activity (A-Z)' },
+    { value: '3_descending', text: 'Activity (Z-A)' },
+  ],
+  offenders: offenderData
+    .sort((a, b) => a.offenderName.localeCompare(b.offenderName, 'en', { ignorePunctuation: true }))
+    .map(data => {
+      const quickLookUrl = `${config.app.notmEndpointUrl}offenders/${data.offenderNo}/quick-look`
+
+      // Return the data in the appropriate format to seed the table macro
+      return [
+        {
+          html: data.location ? `<a href=${quickLookUrl} target="_blank">${data.offenderName}</a>` : data.offenderName,
+        },
+        {
+          text: data.offenderNo,
+        },
+        {
+          text: data.location || '--',
+        },
+        {
+          text: data.activity,
+        },
+        {
+          text: data.comments,
+        },
+      ]
+    }),
+})
+
+const validateDates = dates => {
+  const errors = []
+
+  const fromDate = moment(dates.fromDate, 'DD/MM/YYYY')
+  const toDate = moment(dates.toDate, 'DD/MM/YYYY')
+
+  const now = moment()
+  if (fromDate.isAfter(now, 'day'))
+    errors.push({ text: 'Select a date range that is not in the future', href: '#fromDate' })
+
+  if (fromDate.isAfter(now, 'day'))
+    errors.push({ text: 'Select a date range that is not in the future', href: '#toDate' })
+
+  if (toDate) {
+    const weeks = fromDate.diff(toDate, 'week')
+
+    if (Math.abs(weeks) > 2)
+      errors.push({ text: 'Select a date range that does not exceed two weeks', href: '#fromDate' })
+  }
+  return errors
+}
+
+const dateRangeForStats = (subtractWeeks = 0) => {
+  const sunday = moment()
+    .subtract(subtractWeeks, 'week')
+    .day(0)
+    .format('DD/MM/YYYY')
+  const saturday = moment()
+    .subtract(subtractWeeks, 'week')
+    .day(6)
+    .format('DD/MM/YYYY')
+
+  return {
+    fromDate: sunday,
+    toDate: saturday,
+  }
+}
+
+const formatDatesForDisplay = ({ fromDate, toDate }) => {
+  if (fromDate && toDate) {
+    if (fromDate !== toDate)
+      return `${readableDateFormat(fromDate, 'D/MM/YYYY')} to ${readableDateFormat(toDate, 'D/MM/YYYY')}`
+  }
+
+  return `${readableDateFormat(fromDate, 'D/MM/YYYY')}`
+}
+
+const getStatPresetsLinks = ({ activeCaseLoadId }) => {
+  const weeklyDateRange = dateRangeForStats()
+  const previousWeeklyDateRange = dateRangeForStats(1)
+  const statsFor2WeeksAgoDateRange = dateRangeForStats(2)
+
+  const statsForCurrentWeek = `/manage-prisoner-whereabouts/attendance-reason-statistics?agencyId=${activeCaseLoadId}&period=AM_PM&fromDate=${
```

Comment:
Might be neater if `/manage-prisoner-whereabouts/attendance-reason-statistics?agencyId=${activeCaseLoadId}&period=AM_PM` or at least `/manage-prisoner-whereabouts/attendance-reason-statistics` wasn't repeated.

Reply:
Agreed! Will sort it now.